### PR TITLE
feat: Export `near-primtives` from `near-indexer-primtives`

### PR DESCRIPTION
**Pull request by morgsmccauley**
This PR re-exports the whole `near-primitives` from `near-indexer-primitives` as well as the existing `views` and `types`.

The main motivation behind this is to gain access to the `DelegateAction` types, which are not available through either `views`/`types`.

### CODE REVIEW

**Review by khorolets — APPROVED** [review]

**Review by wacban — APPROVED** [review]
lgtm

**Comment by jakmeier** [comment]
> The main motivation behind this is to gain access to the DelegateAction types, which are not available through either views/types.

Would depending on `near-primitives` directly also work for this use case? Is using the re-exported version preferred for some reason?

**Comment by khorolets** [comment]
> > The main motivation behind this is to gain access to the DelegateAction types, which are not available through either views/types.
> 
> Would depending on `near-primitives` directly also work for this use case? Is using the re-exported version preferred for some reason?

It will work as a workaround, however, we're trying to give a nice developer experience and let developers avoid +1 dependency. We achieve it by re-exporting necessary pieces and this time missed the `DelegateAction`-releated. That's why we've decided to re-export the entire `near-primitives`.

To be clear, it's not a disaster and a workaround is right there, but this change is really nice to have at least in the next NEAR crates release (0.17.0).

**Comment by jakmeier** [comment]
Okay I see, that's a good reason.

IMO, we can certainly include it in the next release. Or even make an extra 0.16.1 release (for all crates). I think you know better than I whether it makes sense to make an extra release or not. My suggestion would be, maybe let's wait a bit longer to see if we have any other changes we need and then check if we want to push another release to crates.io?

Also, I keep saying it, we really need some clarity on who owns each of these crates. I just released them because I was told we need them. But I have no idea who needs these crates and for what, so it is impossible for me to do a proper check on whether a release makes sense and whether the changes make sense. It's always gonna be a bit messy if we keep doing it this way. :)

**Comment by khorolets** [comment]
#### TL;DR
@jakmeier publishing was correct, I missed it.
If it is possible it'd be great to release crates 0.16.1 along with `nearcore` 1.32.0.
Let's do it closer to the 1.32.0 release in case something else to be included appears.

@jakmeier 
> My suggestion would be, maybe let's wait a bit longer to see if we have any other changes we need and then check if we want to push another release to crates.io?

Definitely, we can and should wait, totally agree. It's really kind of you to perform an empty release of a bunch of crates in order to update one of them. But it should be helpful for the developers who build indexers.

> we really need some clarity on who owns each of these crates

Yeah, I feel you. Without some kind of doc/list it's impossible to know who to ping with questions. I recall we used to have a `codeowners` file in the root of the project describing different folders of the repo and who "owns" which crate. Maybe we need something like that yet not strongly required a review so people working on the protocol are not blocked by random dudes like me :)

> I just released them because I was told we need them

Whoah, no one is trying to blame you. You acted absolutely correctly. This particular miss is my bad.

**Comment by jakmeier** [comment]
Thanks @khorolets for the very clear communication!

> Whoah, no one is trying to blame you. You acted absolutely correctly. This particular miss is my bad.

I didn't mean it in that way :) Just kind of pointing out the problem in the process. Someone needs a crate release -> someone else blindly releases X crates. Seems like a very error prone approach that needs flagging.

> If it is possible it'd be great to release crates 0.16.1 along with nearcore 1.32.0.
Let's do it closer to the 1.32.0 release in case something else to be included appears.

@wacban Do you think that's something you can take on your checklist for the 1.32.0 release? Basically just bumping the workspace patch version at some point. See also here: https://near.github.io/nearcore/misc/index.html#crate-versioning-and-publishing
(Or, if you feel extra ambitions, finding a way to do a single crate patch release would be phenomenal! :D)
